src/v/config/validators.cc

```diff
@@ -288,6 +288,40 @@ validate_iceberg_rest_catalog_auth_mode(const config::configuration& config) {
         }
         break;
     }
+    case datalake_catalog_auth_mode::aws_sigv4: {
+        // Check for effective AWS configuration using fallback logic:
+        // prefer iceberg-specific, fallback to cloud storage
+        auto effective_access_key
+          = config.iceberg_rest_catalog_aws_access_key().has_value()
+              ? config.iceberg_rest_catalog_aws_access_key()
+              : config.cloud_storage_access_key();
+        auto effective_secret_key
+          = config.iceberg_rest_catalog_aws_secret_key().has_value()
+              ? config.iceberg_rest_catalog_aws_secret_key()
+              : config.cloud_storage_secret_key();
+        auto effective_region
+          = config.iceberg_rest_catalog_aws_region().has_value()
+              ? config.iceberg_rest_catalog_aws_region()
+              : config.cloud_storage_region();
+
+        if (!(effective_region.has_value() && effective_access_key.has_value()
```

Comment:
Hm. These shouldn't be required if auth mode if credential source (for cloud or the catalog override) is `model::cloud_credentials_source::aws_instance_metadata`.

Related: We can use aws_sigv4 only for aws_instance_metadata and for config file. In case you want to also validate this.